.gitignore

```diff
@@ -50,3 +50,7 @@ packages/cli/test/functional/*/_site
 # vscode configuration
 .vscode/
 *.iml
+
+# Compiled files from TypeScript
+*.js.map
```

Comment:
> The .js files are still kept in repository for easier transitional change, but after this PR those files will slowly be removed and excluded.

I think we should directly *rename* the files for the following reasons:
- duplicating wipes `git blame/annotate` information
- the diff in the PR becomes much more reviewable (from what I can see, most of the files seem to be duplicates, but there are some minor differences like in the test files which are very difficult to spot) and valuable

> easier transitional change

I'm not getting this part though, let me know if I'm missing something 🤔

Reply:
The decision to duplicate is based from https://github.com/MarkBind/markbind/issues/1421#issuecomment-1055225798. Originally I thought to just swap (rename) the `.js` to `.ts`, but maybe in the interest of reducing merge conflicts on other PRs as raised in the discussion, the `.js` files are still kept for now, and would slowly be deleted as the migration goes.

However, I also agree with your point on preserving git annotates, and I think workflow wise it should be better to just swap it as `.ts` so we don't have to keep maintaining the `.js` and `.ts` to perfectly match. Would it be okay for me to just rename them?

Reply:
Agree with this. I think renaming is better as well for the reasons above. This will also be less confusing when working on the core packages (whether to make changes in the `.js` or `.ts` files).

Reply:
> The decision to duplicate is based from https://github.com/MarkBind/markbind/issues/1421#issuecomment-1055225798.

I see, I actually meant leveraging typescript's js inter-op (much like you're doing for the patched files currently) if possible.

- rename one/a few starter files to `.ts` to test tsc in this PR
- the rest in subsequent PRs (where we can even proceed with the typescript code conversion too)

This reduces conflicts with other PRs since we're still dealing with the same `.js` files. Duplicating won't help here as the updates from other PRs need to be transferred to the `.ts` files here constantly (and vice versa in other PRs once this was merged).

... This comment was in the middle of the semester though, (thankfully) not much PRs left touching the core code at this juncture so I don't think we need to worry about this too much. 😌 

> Some files are not yet duplicated to .ts, would like some assistance on whether we should do it or keep it as-is:

Renaming all at once sounds good too, especially since you've already given quite some thought to which files should be converted.

> Would it be okay for me to just rename them?

Let's do that 👍

Reply:
Ahh I understand your point. Sorry that I read it a bit differently 😅 

This makes me rethink my approach though. My current strategy is to rename everything and fix the errors in separate PRs as we go along. While it's easier to fully change the environment first and adapt to it incrementally, this leaves us with issues to handle in the middle of the adaptation (from files riddled with red squiggles, build does not give a zero exit code, publishing concerns, etc).

I think from reading your point, I found a different way to approach it. In separate PRs, we incrementally rename the files _and_ fix the errors, so we make sure that it's successfully compiled without errors and avoiding the issues we have here.

I feel like I would revise the strategy to that, so this PR sets up the workflow and also converts (and fixes) some starter files, like the tests and utils. Would this be a better way to go for this?

(cc @jonahtanjz so everyone is on the same page)